src/runtime_src/core/common/info_memory.cpp

```diff
@@ -593,6 +593,7 @@ populate_hardware_context(const xrt_core::device* device)
     }
 
     try {
+      hw_context.context_status = xrt_core::device_query<xq::device_status>(device);
```

Comment:
The query is for device_status, the result in context_status, which is which and what is what?

Reply:
I will add a comment explaining why. For management devices the device status node states if a device needs to be reset or not due to a hanging kernel. However, a device can only hang when it is has some xclbin loaded! This clearly ties into having the hardware context have some sort of 'hanging' status. For legacy devices this ends up with the context status of the hardware context being the device status! For future devices the status will be returned through the hw_context query and this node will be used for the management partitions.